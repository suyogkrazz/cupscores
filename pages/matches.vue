<template>
  <div>
    <v-layout row wrap>
       <v-flex xs4 offset-xs4 sm4 offset-sm4>
          <v-text-field v-model="search" solo :label="$t('match.search')" ></v-text-field>
        </v-flex>
    </v-layout>
    <v-dialog v-model="matchDialog" max-width="500px">
        <v-card v-if="selectedMatch" class="pa-4">
          <stats :match="selectedMatch"></stats>
          <v-card-actions>
            <v-btn color="primary" flat @click.stop="matchDialog=false">Close</v-btn>
          </v-card-actions>
        </v-card>
    </v-dialog>    
    <v-layout row wrap>
          <v-flex v-for="match in filteredMatches" :key="match.fifa_id"  xs4 sm4 class="pa-4"  @click.stop="openDetails(match)">
                  <v-card>
                      <v-card-title primary-title >
                          <v-layout row align-center justify-space-around>
                              <v-flex  x4> 
                                  <img  :src="getCountryImage(match.away_team.code)" class="icon-image" />
                                  <div > {{match.away_team.country}}</div>
                              </v-flex>
                              <v-flex  x3> 
                                  <h1 :class="{ 'grey--text': (match.away_team.country != match.winner) }">{{match.away_team.goals}}</h1>
                              </V-flex>    
                              <v-flex  x1 style="text-align:center"> 
                                  <h1>-</h1>  
                              </V-flex>    
                              <v-flex  x3 style="text-align:right"> 
                                  <h1 :class="{ 'grey--text': (match.home_team.country != match.winner)}">{{match.home_team.goals}}</h1>
                              </V-flex>    
                              <v-flex  x4 style="text-align:right"> 
                                  <img  :src="getCountryImage(match.home_team.code)" class="icon-image" />
                                  <div> {{match.home_team.country}}</div>
                              </v-flex>
                          </v-layout>
                      </v-card-title>
                  </v-card>
          </v-flex>
    </v-layout>
  </div>
</template>

<script lang="ts">
import { Component, Vue } from "nuxt-property-decorator";
import { State } from "vuex-class";
import { Route } from "vue-router";
import { getCountryImg } from "~/plugins/country_code.js";
import Stats from "~/components/Stats.vue";
@Component({
  components: {
    Stats
  }
})
export default class extends Vue {
  @State matches;
  selectedMatch: any = null;
  matchDialog: boolean = false;
  search: string = "";
  fetch({ store }) {
    store.dispatch("getAllMatch");
  }
  get filteredMatches() {
    if (!this.search) {
      return this.matches;
    }
    return this.matches.filter(item => {
      return (
        item.away_team.country
          .toLowerCase()
          .indexOf(this.search.toLowerCase()) > -1 ||
        item.home_team.country
          .toLowerCase()
          .indexOf(this.search.toLowerCase()) > -1
      );
    });
  }
  openDetails(match) {
    this.selectedMatch = match;
    this.matchDialog = true;
  }

  getCountryImage(iso) {
    return getCountryImg(iso);
  }
}
</script>
<style>
.icon-image {
  width: 45px;
  height: 45px;
}
</style>
