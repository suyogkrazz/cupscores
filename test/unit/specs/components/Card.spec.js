import { shallow } from "@vue/test-utils";
import { createRenderer } from "vue-server-renderer";
import Component from "~/components/Card.vue";
import matches from "../../../data/matches.js";
describe(`Component Logo`, () => {
  it("renders to a snapshot", () => {
    const renderer = createRenderer();
    const match = matches[0];
    const wrapper = shallow(Component, {
      propsData: {
        match,
        team: "away"
      }
    });
    renderer.renderToString(wrapper.vm, (err, str) => {
      if (err) throw new Error(err);
      expect(str).toMatchSnapshot();
    });
  });
});
