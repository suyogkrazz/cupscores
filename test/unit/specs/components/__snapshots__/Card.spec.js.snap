// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component Logo renders to a snapshot 1`] = `
<v-flex sm4="">
    <v-card class="elevation-4">
        <v-card-media height="200px" src="http://flagpedia.net/data/flags/normal/sa.png" class="white--text"></v-card-media>
        <v-card-title primary-title="" style="display:flex;flex-wrap:wrap;justify-content:center;align-items:center;">
            <div>
                <h3 class="headline mb-0">Saudi Arabia</h3>
            </div>
        </v-card-title>
        <v-card-title primary-title="" style="display:flex;flex-wrap:wrap;justify-content:center;align-items:center;">
            <div>
                <h3 class="headline mb-0" style="font-size:45px!important;color:blue;text-align:left;">0</h3>
            </div>
        </v-card-title>
    </v-card>
</v-flex>
`;
